src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -0,0 +1,194 @@
+    }
+
+    private void doSetup() throws IOException, CommandException
+    {
+        _containerHelper.createProject(getProjectName(), null);
+        _containerHelper.createProject(LINKED_STUDY_PROJECT, "Study");
+        clickButton("Create Study");
+        clickButton("Create Study");
+
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+        new PortalHelper(getDriver()).addBodyWebPart("Sample Types");
+
+        goToProjectHome(LINKED_STUDY_PROJECT);
+        new PortalHelper(getDriver()).addBodyWebPart("Datasets");
+
+        createSampleTypes();
+    }
+
+    private void createSampleTypes() throws IOException, CommandException
+    {
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+
+        log("Creating sample types");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", SAMPLE_TYPE1);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String)));
+        dgen.createDomain(createDefaultConnection(), SAMPLE_TYPE_DOMAIN_KIND);
+        dgen.addCustomRow(Map.of("name", "blood"));
+        dgen.addCustomRow(Map.of("name", "urine"));
+        dgen.addCustomRow(Map.of("name", "stool"));
+        dgen.addCustomRow(Map.of("name", "sweat"));
+        dgen.insertRows(createDefaultConnection(), dgen.getRows());
+
+        lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", SAMPLE_TYPE2);
+        dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                        new FieldDefinition("int", FieldDefinition.ColumnType.Integer)));
+        dgen.createDomain(createDefaultConnection(), SAMPLE_TYPE_DOMAIN_KIND);
+        dgen.addCustomRow(Map.of("name", "first"));
+        dgen.addCustomRow(Map.of("name", "second"));
+        dgen.addCustomRow(Map.of("name", "third"));
+        dgen.addCustomRow(Map.of("name", "fourth"));
+        dgen.insertRows(createDefaultConnection(), dgen.getRows());
+
+    }
+
+    @Test
+    public void testLinkToStudy()
+    {
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+
+        log("Linking sample types one row to study");
+        linkToStudy(SAMPLE_TYPE1, 2, Arrays.asList("P1", "P2"), Arrays.asList("1", "2"));
+
+        log("Verifying the linked sample type in study");
+        goToProjectHome(LINKED_STUDY_PROJECT);
+        clickAndWait(Locator.linkWithText(SAMPLE_TYPE1));
+        DataRegionTable table = new DataRegionTable("Dataset", getDriver());
+
+        checker().verifyEquals("Incorrect number of rows linked", 2, table.getDataRowCount());
+        checker().verifyEquals("Incorrect Participant ID's", Arrays.asList("P1", "P2"), table.getColumnDataAsText("ParticipantId"));
+
+        log("Verifying log entries");
```

Comment:
It might also be beneficial to ensure that we can link the same sample to multiple studies and verify that the links are rendered correctly.

Reply:
Done